frontend/catalyst/jax_primitives_utils.py

```diff
@@ -45,28 +70,35 @@ def get_call_equation(jaxpr):
     raise AssertionError("No call_jaxpr found in the JAXPR.")
 
 
-def lower_jaxpr(ctx, jaxpr, context=None):
+def lower_jaxpr(ctx, jaxpr, metadata=None, fn=None):
     """Lowers a call primitive jaxpr, may be either func_p or quantum_kernel_p
 
     Args:
         ctx: LoweringRuleContext
         jaxpr: JAXPR to be lowered
-        context: additional context to distinguish different FuncOps
+        metadata: additional metadata to distinguish different FuncOps
 
     Returns:
         FuncOp
     """
-    equation = get_call_equation(jaxpr)
-    call_jaxpr = equation.params["call_jaxpr"]
-    callable_ = equation.params.get("fn")
-    if callable_ is None:
-        callable_ = equation.params.get("qnode")
-    pipeline = equation.params.get("pipeline")
-    return lower_callable(ctx, callable_, call_jaxpr, pipeline=pipeline, context=context)
+
+    if fn is None or isinstance(fn, qml.QNode):
+        equation = get_call_equation(jaxpr)
+        call_jaxpr = equation.params["call_jaxpr"]
+        pipeline = equation.params.get("pipeline")
+        callable_ = equation.params.get("fn")
+        if callable_ is None:
+            callable_ = equation.params.get("qnode", None)
+    else:
```

Comment:
What's the purpose of the `fn` parameter and this branch?

Reply:
Hmm, good question. It's hard to say I think that code has gone through a few iterations (like the cache on Function that is now gone). If you found a way to simplify it then I think that's great!